packages/commonwealth/client/scripts/controllers/app/webWallets/cosmos_evm_metamask_web_wallet.ts

```diff
@@ -52,28 +55,35 @@ class CosmosEvmWebWalletController implements IWebWallet<string> {
     return this._accounts || [];
   }
 
-  public async signMessage(message: string): Promise<string> {
+  public async getRecentBlock(chainIdentifier: string) {
+    const url = `${window.location.origin}/cosmosAPI/${
+      chainIdentifier
+    }`;
+    const client = await StargateClient.connect(url);
+    const height = await client.getHeight();
+    const block = await client.getBlock(height);
+
+    return {
+      number: block.header.height,
+      hash: block.id,
+      // seconds since epoch
+      timestamp: Math.floor((new Date(block.header.time)).getTime() / 1000)
+    };
+  }
```

Comment:
Canvas just uses the block hash to verify that a session/action request was made after a certain timestamp, so we can get away with just using Ethereum blocks regardless of the wallet/chain that is being used to log in.